src/monitor/group_state_machine.c

```diff
@@ -431,12 +432,17 @@ ProceedGroupState(AutoFailoverNode *activeNode)
 		LogAndNotifyMessage(
 			message, BUFSIZE,
 			"Setting goal state of " NODE_FORMAT
-			" to secondary after it caught up.",
-			NODE_FORMAT_ARGS(activeNode));
+			" to secondary after it caught up and " NODE_FORMAT
+			" to apply_settings to add the secondary to synchronous_standby_names",
+			NODE_FORMAT_ARGS(activeNode),
+			NODE_FORMAT_ARGS(primaryNode));
 
 		/* node is ready for promotion */
 		AssignGoalState(activeNode, REPLICATION_STATE_SECONDARY, message);
 
+		/* edit synchronous_standby_names to add the new standby now */
+		AssignGoalState(primaryNode, REPLICATION_STATE_APPLY_SETTINGS, message);
```

Comment:
This should not be done when the current primary is in state wait_primary, because it still has to wait to be sure that WAL has propagated to the secondary. This is done in the transition from wait_primary->primary but not from wait_primary->apply_settings. Once the primary is in state apply_settings/apply_settings, we consider it a full primary and a failover is allowed.

Reply:
also the comment at the top of this if statement seems like it might be a bit outdated.

Reply:
Should be fixed now, thanks!